### PR TITLE
Metadata snippets

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

**Comment by evildour** [comment]
Rebased with latest changes

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+namespace Google.Cloud.Metadata.V1.IntegrationTests
+{
+    /// <summary>
+    /// Base class for a fixture which is set up at the start of the test run, and torn down at the end.
+    /// </summary>
+    public class MetadataFixtureBase : IDisposable
+    {
+        private const string EmulatorEnvironmentVariable = "METADATA_EMULATOR_HOST";
+
+        private Process emulatorProcess;
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: Make these readonly?

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.Snippets/MetadataClientSnippets.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+using Newtonsoft.Json.Linq;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Metadata.V1.Snippets
+{
+    [Collection(nameof(MetadataSnippetFixture))]
+    public class MetadataClientSnippets
+    {
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.Snippets/MetadataClientSnippets.cs]: I'd normally expect a constructor taking a MetadataSnippetFixture here, setting in as a field.
I'm slightly curious that this works, but I believe you :)

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs (hunk 1 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: Done

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.Snippets/MetadataClientSnippets.cs (hunk 2 above)
Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.Snippets/MetadataClientSnippets.cs]: Added just in case (to the integration tests as well)

**Pull request merged**